internal/utils/array.go

```diff
@@ -0,0 +1,15 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package utils
+
+// Flatten takes as argument a nested array of elements T.
+// Example: [["a", "b"], ["c", "d"]] -> ["a", "b", "c", "d"]
+func Flatten[T any](nested [][]T) []T {
```

Comment:
Move this to `core/watcher/watchertest`, if it becomes useful we can extract then.

Reply:
This is even better to put in _juju/collections_ in the `transform` package.